src/hdf5/detail/h5/h5_file.hpp

```diff
@@ -154,13 +154,47 @@ class HighFiveFile
     }
 
 
-
+    /*
+     * Write attribute on all mpi cores, considered global, always the same path/key and value
+     */
     template<typename Data>
-    void write_attribute(std::string const& path, std::string const& key, Data const& value)
+    void write_attribute(std::string const& keyPath, std::string const& key, Data const& data)
     {
-        h5file_.getGroup(path)
-            .template createAttribute<Data>(key, HighFive::DataSpace::From(value))
-            .write(value);
+        // assumes all keyPaths and values are identical, and no null patches
+        // clang-format off
+        PHARE_DEBUG_DO(
+            auto paths = core::mpi::collect(keyPath, core::mpi::size());
+            for (auto const& path : paths)
+                PHARE_LOG_LINE_STR(std::to_string(core::mpi::size()) << " " << path)
+            if (!core::all(paths, [&](auto const& path) { return path == paths[0]; }))
+                throw std::runtime_error("Function does not support different paths per mpi core");
+        )
+        // clang-format on
+
+        constexpr bool data_is_vector = core::is_std_vector_v<Data>;
```

Comment:
#include "core/utilities/types.hpp"
#include "core/utilities/meta/meta_utilities.hpp" (for is_iterable_v not in this diff)